Description: Fix build failures on arm platforms
Author: Stefan Bader <stefan.bader@canonical.com>
Index: xen-4.9.0/tools/libxc/xc_dom_arm.c
===================================================================
--- xen-4.9.0.orig/tools/libxc/xc_dom_arm.c
+++ xen-4.9.0/tools/libxc/xc_dom_arm.c
@@ -223,6 +223,7 @@ static int set_mode(xc_interface *xch, d
 
     domctl.domain = domid;
     domctl.cmd    = XEN_DOMCTL_set_address_size;
+    domctl.u.address_size.size = 0;
     for ( i = 0; i < ARRAY_SIZE(types); i++ )
         if ( !strcmp(types[i].guest, guest_type) )
             domctl.u.address_size.size = types[i].size;
