From: Bastian Blank <waldi@debian.org>
Date: Sat, 5 Jul 2014 11:46:59 +0200
Subject: tools-misc-prefix.diff

Patch-Name: tools-misc-prefix.diff

---

--- a/tools/misc/Makefile
+++ b/tools/misc/Makefile
@@ -54,12 +54,8 @@ all build: $(TARGETS_BUILD)
 
 .PHONY: install
 install: build
-	$(INSTALL_DIR) $(DESTDIR)$(bindir)
-	$(INSTALL_DIR) $(DESTDIR)$(sbindir)
 	$(INSTALL_DIR) $(DESTDIR)$(LIBEXEC_BIN)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_BIN) $(DESTDIR)$(bindir)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_SBIN) $(DESTDIR)$(sbindir)
-	$(INSTALL_PYTHON_PROG) $(INSTALL_PRIVBIN) $(DESTDIR)$(LIBEXEC_BIN)
+	$(INSTALL_PYTHON_PROG) $(INSTALL_BIN) $(INSTALL_SBIN) $(INSTALL_PRIVBIN) $(DESTDIR)$(LIBEXEC_BIN)
 
 .PHONY: clean
 clean:
